shared/src/commonMain/kotlin/com/mirego/kmp/boilerplate/datasource/apollo/ApolloGraphQLDiskDataSource.kt

```diff
@@ -0,0 +1,71 @@
+package com.mirego.kmp.boilerplate.datasource.apollo
+
+import com.apollographql.apollo3.api.CustomScalarAdapters
+import com.apollographql.apollo3.api.Query
+import com.apollographql.apollo3.api.json.BufferedSinkJsonWriter
+import com.apollographql.apollo3.api.json.BufferedSourceJsonReader
+import com.mirego.kmp.boilerplate.app.coroutines.Dispatchers
+import com.mirego.kmp.boilerplate.app.filesystem.NativeFileSystem
+import com.mirego.trikot.datasources.flow.BaseExpiringExecutableFlowDataSource
+import com.mirego.trikot.datasources.flow.FlowDataSourceExpiringValue
+import kotlinx.coroutines.withContext
+import kotlinx.serialization.SerializationException
+import okio.Path.Companion.toPath
+import okio.buffer
+
+class ApolloGraphQLDiskDataSource<R : ApolloGraphQLDataSourceRequest<T>, T : Query.Data>(
```

Comment:
I think we should instead use Apollo's built-in [normalized cache](https://www.apollographql.com/docs/kotlin/caching/normalized-cache) instead of implementing our own caching solution.

Reply:
Does any of our projects use the Apollo cache now? I now we had issues with it in the past

Reply:
In transat, we used normalized cache in the beginning but switched to Trikot datasources afterwards. If I remember correctly, there were 2 reasons:
1) Noticeable performance issues (might be fixed now I suppose, since this was in mid-end of 2022)
2) There were no easy way to manage states (Loading, Error, Data)

I feel like, for now, we should stick to Trikot datasources in the boilerplate until more projects uses normalized cache (and can vouch that everything works great!)